rkt/prepare.go

```diff
@@ -72,6 +77,12 @@ func runPrepare(cmd *cobra.Command, args []string) (exit int) {
 		}
 	}
 
+	if flagPrivateUsers && common.SupportsUserNS() {
+		rand.Seed(time.Now().UnixNano())
+		uidshift := rand.Intn(0xffff)
```

Comment:
So here you're generating a number in the interval `[0x0000,0xFFFF)` (note the interval is closed at the start and open at the end) and then from this number you assign `0x10000 (65536)` UIDs to the container.

That means that the lowest range you'll get is `[0x0000-0xFFFF]` (no shifting) and the highest range is `[0xFFFE-1FFFD]`.

I think we should generate a number in the range `(0x0000FFFF-0xFFFF0000]` (note the interval is open at the start and closed at the end) and then the lowest range would be `[0x00010000-0x0001FFFF]` and the highest `[0xFFFF0000-0xFFFFFFFF]`.

Does that make sense?